gensim/matutils.py

```diff
@@ -382,6 +383,93 @@ def cossim(vec1, vec2):
+            dense2 = sparse2full(vec2,max_len)
+            return scipy.stats.entropy(dense1,dense2)
+        else:
+            return scipy.stats.entropy(vec1,vec2)
+    elif type(vec1) is list and type(vec2) is list:
+        max_len = max(len(vec1),len(vec2),num_of_features)
+        dense1 = sparse2full(vec1,max_len)
+        dense2 = sparse2full(vec2,max_len)
+        return scipy.stats.entropy(dense1,dense2)
+
+
+def hellinger(vec1, vec2):
+    """
+    Hellinger distance is a distance metric to quanitfy the similarity between two probability distributions.
+    Similarity between distributions will be a number between <0,1>, where 0 is maximum similarity and 1 is minimum similarity.
+    """
+    if scipy.sparse.issparse(vec1) and scipy.sparse.issparse(vec2):
+        dense1 = vec1.todense()
+        dense2 = vec2.todense()
+        # if it is a sparse matrix in bag of words format, convert to list and go ahead.
+        if dense1.shape[1] == 2 and dense2.shape[1] == 2:
+            dense1,dense2 = dict(dense1.tolist()),dict(dense2.tolist())
+            if len(dense2) < len(dense1):
+                dense1, dense2 = dense1, dense2 # swap references so that we iterate over the shorter vector
+            sim = numpy.sqrt(0.5*sum((numpy.sqrt(value) - numpy.sqrt(dense2.get(index, 0.0)))**2 for index, value in iteritems(dense1)))
+            return sim
+        # else if it is in a different format, directly go about it
+        sim = numpy.sqrt(0.5 * ((numpy.sqrt(dense1) - numpy.sqrt(dense2))**2).sum())
+        return sim
+    elif isinstance(vec1,numpy.ndarray) and isinstance(vec2,numpy.ndarray):
+        # if it is in a bag of words format, go ahead with list method
+        if vec1.shape[1] == 2 and vec2.shape[1] == 2:
+            vec1, vec2 = dict(vec1.tolist()), dict(vec2.tolist())
+            if len(vec2) < len(vec1):
+                vec1, vec2 = vec2, vec1
+            sim = numpy.sqrt(0.5*sum((numpy.sqrt(value) - numpy.sqrt(vec2.get(index, 0.0)))**2 for index, value in iteritems(vec1)))
+            return sim
+        # else if it is in a different format, directly go about it
+        sim = numpy.sqrt(0.5 * ((numpy.sqrt(vec1) - numpy.sqrt(vec2))**2).sum())
+        return sim
+    elif type(vec1) is list and type(vec2) is list:
+        vec1, vec2 = dict(vec1), dict(vec2)
+        if len(vec2) < len(vec1):
+            vec1, vec2 = vec2, vec1
+        sim = numpy.sqrt(0.5*sum((numpy.sqrt(value) - numpy.sqrt(vec2.get(index, 0.0)))**2 for index, value in iteritems(vec1)))
+        return sim
+
+
+def jaccard(vec1,vec2):
+    """
+    A similarity metric between bags of words representation.
+    Return the intersection divided by union, where union is the sum of the size of the two bags.
+    Highest value jaccard similarity of two bags is 1/2, which indicates the highest similarity.
+    """
+    intersection = 0
+    union = sum(row[1] for row in vec1) + sum(row[1] for row in vec2)
```

Comment:
`row` is probably a poor choice of word, what I mean to do is to get the value of the `weight` of the corresponding `id` in a bag of words representation which Jaccard makes sense for. What should I change this variable to to make more sense?

The union is meant to be the sum of the sizes of the two bags you are comparing. This is the definition of jaccard similarity according to [this](http://infolab.stanford.edu/~ullman/mmds/ch3.pdf) text. For e.g, The bag-similarity of bags `{a, a, a, b}` and `{a, a, b, b, c}` is `1/3`. The size of the bags is `9` and the size of intersection is `3`. I'd expect these bags to be represented as `[(0,3),(1,1)]`  and `[(0,2),(1,2),(2,1)]`. 
Though now that I think about this more, I think I have to re-write my jaccard method to accept other kinds of inputs and be more generic... Will work on it.